src/main/java/io/nats/client/impl/SocketDataPort.java

```diff
@@ -24,14 +26,28 @@
 import java.net.InetSocketAddress;
 import java.net.Socket;
 import java.net.URI;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
 import java.time.Duration;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 /**
  * This class is not theadsafe.  Caller must ensure thread safety.
  */
 public class SocketDataPort implements DataPort {
+    private static final int MAX_LINE_LEN = 8192;
```

Comment:
I'd be interested in implementing some sort of factory here to get an implementation based on the protocol, instead of just making this class bigger. May not be practical, just something to consider.

Reply:
I think that would be a MUCH better design... I just wanted to provide the minimal changes necessary to get websocket support. More specifically, if the Options class allowed one to add a `DataPort` factory that is keyed off of the "schema", then supporting new transports (such as websockets) might be as simple as defining a new `DataPort` factory.

However, the current `DataPort` abstraction could probably use some work. For example, instead of having the `connect()` method, that should probably be a method on the `DataPort` factory, and instead of passing in a `NatsConnection` it should probably pass in an `Options` object.

On a completely separate side-note, has anyone considered changing the library to use "reactive streams"? ...in which case the code would need to be substantially modified (although the "heart" of the code is really just the WebsocketFrameHeader and the initial hand-shake which is currently in the SocketDataPort.